src/components/CurrencySelector/CurrencyOptions.tsx

```diff
@@ -1,39 +1,73 @@
 import { Combobox } from "@headlessui/react";
-import { Currency } from "./types";
-import { isEmpty } from "helpers";
+import { Currency } from "types/components";
+import { QueryState } from "types/third-party/redux";
+import QueryLoader from "components/QueryLoader";
 
 type Props = {
   classes?: string;
-  currencies: Currency[];
+  currencies: Currency[] | QueryState<Currency[]>;
```

Comment:
move loading of currency options inside `<CurrencySelector/>` so whole form need not wait for currencies to load - most users could certainly start with USD